candle-transformers/src/models/debertav2.rs

```diff
@@ -0,0 +1,1499 @@
+use std::collections::HashMap;
+
+use candle::{DType, Device, Module, Tensor, D};
+use candle_nn::{
+    conv1d, embedding, layer_norm, Conv1d, Conv1dConfig, Embedding, LayerNorm, VarBuilder,
+};
+use serde::{Deserialize, Deserializer};
+
+pub const DTYPE: DType = DType::F32;
+
+// NOTE: HiddenAct and HiddenActLayer are both direct copies from bert.rs.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum HiddenAct {
+    Gelu,
+    GeluApproximate,
+    Relu,
+}
+
+pub struct HiddenActLayer {
+    act: HiddenAct,
+    span: tracing::Span,
+}
+
+impl HiddenActLayer {
+    fn new(act: HiddenAct) -> Self {
+        let span = tracing::span!(tracing::Level::TRACE, "hidden-act");
+        Self { act, span }
+    }
+
+    fn forward(&self, xs: &Tensor) -> candle::Result<Tensor> {
+        let _enter = self.span.enter();
+        match self.act {
+            // https://github.com/huggingface/transformers/blob/cd4584e3c809bb9e1392ccd3fe38b40daba5519a/src/transformers/activations.py#L213
+            HiddenAct::Gelu => xs.gelu_erf(),
+            HiddenAct::GeluApproximate => xs.gelu(),
+            HiddenAct::Relu => xs.relu(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Default)]
+#[serde(rename_all = "lowercase")]
+enum PositionEmbeddingType {
+    #[default]
+    Absolute,
+}
+
+pub type Id2Label = HashMap<u32, String>;
+pub type Label2Id = HashMap<String, u32>;
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct Config {
```

Comment:
This looks a lot like the normal BERT configuration, but in going through the Python Transformers code I realized there were a lot of other configuration tidbits that didn't exist in other BERT models. So, I just started over with a new one just for Deberta.